addons/cluster-autoscaler/v1.4.0.yaml

```diff
@@ -0,0 +1,42 @@
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler
+  labels:
+    k8s-app: cluster-autoscaler
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: cluster-autoscaler
+  template:
+    metadata:
+      labels:
+        k8s-app: cluster-autoscaler
+    spec:
+        containers:
+          - image:
+            name: {{IMAGE}}
+            resources:
+              limits:
+                cpu: 100m
+                memory: 300Mi
+              requests:
+                cpu: 100m
+                memory: 300Mi
+            command:
+              - ./cluster-autoscaler
+              - --cloud-provider={{CLOUD_PROVIDER}}
+              - --nodes={{MIN_NODES}}:{{MAX_NODES}}:{{GROUP_NAME}}
+            env:
+              - name: AWS_REGION
```

Comment:
this is only required for AWS, I figured it wouldn't do any harm if its added for GCE also, if its misleading we could have separate addon directories for each cloud provider. 

```bash
kops/addons/cluster-autoscaler/aws/addon.yml
kops/addons/cluster-autoscaler/gce/addon.yml
etc..
```